src/main/java/mate/academy/bookstore/service/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+                .orElseGet(() -> createNewShoppingCart(user));
+        Book book = bookRepository.findById(requestDto.getBookId())
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book with id: "
+                        + requestDto.getBookId()));
+
+        Optional<CartItem> existingCartItem = cartItemRepository
+                .findByShoppingCartIdAndBookId(shoppingCart.getId(), book.getId());
+
+        if (existingCartItem.isPresent()) {
+            CartItem cartItem = existingCartItem.get();
+            cartItem.setQuantity(cartItem.getQuantity() + requestDto.getQuantity());
+            cartItemRepository.save(cartItem);
+        } else {
+            CartItem newCartItem = new CartItem();
+            newCartItem.setShoppingCart(shoppingCart);
+            newCartItem.setBook(book);
+            newCartItem.setQuantity(requestDto.getQuantity());
+            cartItemRepository.save(newCartItem);
+            shoppingCart.getCartItems().add(newCartItem);
+        }
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Transactional
+    @Override
+    public ShoppingCartDto updateQuantity(Long cartItemId, UpdateCartItemRequestDto requestDto) {
+        CartItem cartItem = cartItemRepository.findById(cartItemId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find cart item with id: "
+                        + cartItemId));
+        cartItem.setQuantity(requestDto.getQuantity());
+        cartItemRepository.save(cartItem);
+        return shoppingCartMapper.toDto(cartItem.getShoppingCart());
+    }
+
+    @Transactional
+    @Override
+    public ShoppingCartDto removeBookFromShoppingCart(Long cartItemId) {
+        User user = getCurrentUser();
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUser(user)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find shopping cart for user"));
+
+        Optional<CartItem> cartItemToRemove = shoppingCart.getCartItems().stream()
+                .filter(item -> item.getId().equals(cartItemId))
+                .findFirst();
+
+        if (cartItemToRemove.isPresent()) {
+            shoppingCart.getCartItems().remove(cartItemToRemove.get());
+            shoppingCartRepository.save(shoppingCart);
+            return shoppingCartMapper.toDto(shoppingCart);
+        } else {
+            throw new EntityNotFoundException("Can't find cart item with id: "
+                    + cartItemId + " in your cart");
+        }
+    }
+
+    private User getCurrentUser() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication != null && authentication.getPrincipal() instanceof User) {
+            return (User) authentication.getPrincipal();
+        }
+        throw new RuntimeException("Can't get current user");
+    }
+
+    private ShoppingCart createNewShoppingCart(User user) {
```

Comment:
this method should be in UserService, let's create shop cart for user during registration
```suggestion
    @Override
    public ShoppingCart createNewShoppingCart(User user) {
```